tests/cnf/core/network/sriov/tests/qinq.go

```diff
@@ -663,3 +814,130 @@ func disableQinQOnSwitch(switchCredentials *sriovenv.SwitchCredentials, switchIn
+		NetConfig.DpdkTestContainer).WithSecondaryNetwork(serverPodNetConfig).
+		DefineOnNode(nodeName).RedefineDefaultContainer(*dpdkContainerCfg).WithHugePages()
+	configMapData := map[string]string{"cmd_file": dpdkPort0Cmd}
+	configMap, err := configmap.NewBuilder(APIClient, "dpdk-port-cmd", tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create config map")
+	dpdkPod.WithLocalVolume(configMap.Definition.Name, "/etc/cmd")
+	dpdkPod, err = dpdkPod.CreateAndWaitUntilRunning(10 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Fail to create server pod")
+
+	return dpdkPod
+}
+
+func defineAndCreateClientDPDKPod(
+	podName,
+	nodeName string,
+	serverPodNetConfig []*multus.NetworkSelectionElement) *pod.Builder {
+	var rootUser int64 = 0
+	securityContext := corev1.SecurityContext{
+		RunAsUser: &rootUser,
+		Capabilities: &corev1.Capabilities{
+			Add: []corev1.Capability{"IPC_LOCK", "SYS_RESOURCE", "NET_RAW"},
+		},
+	}
+
+	dpdkContainer := pod.NewContainerBuilder(podName, NetConfig.DpdkTestContainer,
+		[]string{"/bin/bash", "-c", "sleep INF"})
+	dpdkContainerCfg, err := dpdkContainer.WithSecurityContext(&securityContext).
+		WithResourceLimit("2Gi", "1Gi", 4).
+		WithResourceRequest("2Gi", "1Gi", 4).WithEnvVar("RUN_TYPE", "testcmd").
+		GetContainerCfg()
+	Expect(err).ToNot(HaveOccurred(), "Fail to set dpdk container")
+
+	dpdkPod := pod.NewBuilder(APIClient, podName, tsparams.TestNamespaceName,
+		NetConfig.DpdkTestContainer).WithSecondaryNetwork(serverPodNetConfig).
+		DefineOnNode(nodeName).RedefineDefaultContainer(*dpdkContainerCfg).WithHugePages()
+
+	dpdkPod, err = dpdkPod.CreateAndWaitUntilRunning(10 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Fail to create server pod")
+
+	return dpdkPod
+}
+
+func rxTrafficOnClientPod(clientPod *pod.Builder, clientRxCmd string) {
+	Expect(clientPod.WaitUntilRunning(time.Minute)).ToNot(HaveOccurred(),
+		"Fail to wait until pod is running")
+
+	clientOut, err := clientPod.ExecCommand([]string{"/bin/bash", "-c", clientRxCmd})
+
+	var timeoutError = "command terminated with exit code 137"
+	if err.Error() != timeoutError {
+		Expect(err).ToNot(HaveOccurred(), "Fail to exec cmd")
+	}
+
+	By("Parsing output from the DPDK application")
+	glog.V(90).Infof("Processing testpdm output from client pod \n%s", clientOut.String())
+	Expect(checkRxOnly(clientOut.String())).Should(BeTrue(),
+		"Fail to process output from dpdk application")
+}
+
+func checkRxOnly(out string) bool {
+	lines := strings.Split(out, "\n")
+	Expect(len(lines)).To(BeNumerically(">=", 3),
+		"Fail line list contains less than 3 elements")
+
+	for i, line := range lines {
+		if strings.Contains(line, "NIC statistics for port") {
+			if len(lines) > i && getNumberOfPackets(lines[i+1], "RX") > 0 {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func getNumberOfPackets(line, firstFieldSubstr string) int {
```

Comment:
Could you add comments or example of the output?

Reply:
I can manually run and take from the output but I dont think it will be very helpful.

Reply:
No problem, if you're sure about the functions getNumberOfPackets and checkRxOnly. I cannot review them without an example.

Reply:
Hi Evgeny here is the output from the client command 
The out is separated into lines. Each line should have more then 3 elements. 
The getNumberOfPackets the checks that the RX value is greater then 0. 
Port statistics ====================================
  ######################## NIC statistics for port 0  ########################
  RX-packets: 16         RX-missed: 94636484   RX-bytes:  6813828440
  RX-errors: 0
  RX-nombuf:  0         
  TX-packets: 99861056   TX-errors: 0          TX-bytes:  6790551400

  Throughput (since last show)
  Rx-pps:            2          Rx-bps:  10896278584
  Tx-pps:     19968083          Tx-bps:  10862640120